### PR TITLE
feat: policy-based deployment

### PR DESCRIPTION
**Pull request by ajcasagrande**
## PR Checklist
Please check if your PR fulfills the following requirements:

- **[N/A]** Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-examples/blob/master/.github/CONTRIBUTING.md.

## What is the current behavior?

Pattern only deployment

## What is the new behavior?
Support for policy-based deployments, which allows user to register a node, and then later assign it a workload through node properties.

Also adds env variables for auto authenticating with docker registries when pushing images.

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## New Imports
Are there any new imports or modules? If so, what are they used for and why?

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

MQTT-over-websocket port has been changed from 9001 to 59001 to avoid port conflicts.

### CODE REVIEW

File: demos/OH-EXF-Kamakura/src/app-camera-management/service.policy.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
```
Comment (presatish) [inline demos/OH-EXF-Kamakura/src/app-camera-management/service.policy.json:1]: Need to add the license header as I remember you adding for other json files before.

File: demos/OH-EXF-Kamakura/src/asc-mqtt/service.policy.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (presatish) [inline demos/OH-EXF-Kamakura/src/asc-mqtt/service.policy.json:1]: Missing license header.

File: demos/OH-EXF-Kamakura/src/edgex-ui/service.policy.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (presatish) [inline demos/OH-EXF-Kamakura/src/edgex-ui/service.policy.json:1]: Missing license header

File: demos/OH-EXF-Kamakura/horizon/node-policy.json [hunk 4]
```diff
@@ -1,8 +1,12 @@
 {
```
Comment (presatish) [inline demos/OH-EXF-Kamakura/horizon/node-policy.json:1]: Missing license header.

**Review by presatish — CHANGES_REQUESTED** [review]
Mostly looks goods just some files missing license headers.

File: demos/OH-EXF-Kamakura/src/app-camera-management/service.policy.json (hunk 1 above)
Comment (ajcasagrande) [reply demos/OH-EXF-Kamakura/src/app-camera-management/service.policy.json:1]: @presatish These files are not sent through the json pre-processor, so license header cannot be added. I also do not think it is that important for these files.

**Review by presatish — APPROVED** [review]

**Comment by ajcasagrande** [comment]
@joewxboy

**Review by joewxboy — APPROVED** [review]

**Pull request merged**